src/fetcher.ts

```diff
@@ -60,6 +62,15 @@ export class BaseFetcher {
     this.maxRetries = clientOptions.maxRetries || 2
   }
 
+  public validate(schema: Joi.ObjectSchema, data: any) {
```

Comment:
```suggestion
  public validate<T>(schema: Joi.ObjectSchema, data: T) {
```
we can add this to convert in to generic type